src/zampy/datasets/land_cover.py

```diff
@@ -0,0 +1,247 @@
+
+        data_files = list(ingest_folder.glob(data_file_pattern))
+
+        for file in data_files:
+            # start conversion process
+            print(f"Start processing file `{file.name}`.")
+            ds = xr.open_dataset(file)
+            ds = converter.convert(ds, dataset=self, convention=convention)
+            # TODO: support derived variables
+            # TODO: other calculations
+            # call ds.compute()
+
+        return True
+
+
+def unzip_raw_to_netcdf(
+    ingest_folder: Path,
+    file: Path,
+    overwrite: bool = False,
+) -> None:
+    """Convert a downloaded zip netcdf file to a standard CF/Zampy netCDF file.
+
+    Args:
+        ingest_folder: Folder where the files have to be written to.
+        file: Path to the land cover .zip archive.
+        overwrite: Overwrite all existing files. If False, file that already exist will
+            be skipped.
+    """
+    ncfile = ingest_folder / file.with_suffix(".nc").name
+    if ncfile.exists() and not overwrite:
+        print(f"File '{ncfile.name}' already exists, skipping...")
+    else:
+        ds = extract_netcdf_to_zampy(file)
+        ds.to_netcdf(path=ncfile)
+
+
+def extract_netcdf_to_zampy(file: Path) -> xr.Dataset:
+    """Extract zipped data and convert to zampy format.
+
+    Args:
+        file: Path to the land cover .zip archive.
+
+    Returns:
+        Coarse land cover data in zampy format.
+    """
+    with TemporaryDirectory() as temp_dir:
+        unzip_folder = Path(temp_dir)
+        with ZipFile(file, "r") as zip_object:
+            zipped_file_name = zip_object.namelist()[0]
+            zip_object.extract(zipped_file_name, path=unzip_folder)
+
+        # only keep land cover class variable
+        ds = xr.open_dataset(unzip_folder / zipped_file_name)
+        var_list = [var for var in ds.data_vars]
+        raw_variable = "lccs_class"
+        var_list.remove(raw_variable)
+        ds = ds.drop_vars(var_list)
+
+        # coarsen to fit into memory
+        ds = ds.sortby(["lat", "lon"])
+        ds = ds.rename({"lat": "latitude", "lon": "longitude"})
+        new_grid = xarray_regrid.Grid(
+            north=90,
+            east=180,
+            south=-90,
+            west=-180,
+            resolution_lat=0.25,  # same as resolution of ERA5, must be sufficient
+            resolution_lon=0.25,
+        )
+
+        target_dataset = xarray_regrid.create_regridding_dataset(new_grid)
+
+        ds_regrid = ds.regrid.most_common(target_dataset, time_dim="time", max_mem=1e9)
+
+        # rename variable to follow the zampy convention
+        variable_name = "land_cover"
+        ds_regrid = ds_regrid.rename({raw_variable: variable_name})
+        ds_regrid[variable_name].attrs["units"] = str(
+            VARIABLE_REFERENCE_LOOKUP[variable_name].unit
+        )
+        ds_regrid[variable_name].attrs["description"] = VARIABLE_REFERENCE_LOOKUP[
+            variable_name
+        ].desc
+
```

Comment:
This context manager is trying to clean up at the end, but cannot because `ds` is still open.
```suggestion
ds.close()  # Release the file for clean-up.
```

Reply:
I tried it indeed (see this commit https://github.com/EcoExtreML/zampy/pull/32/commits/abf11dabad7e0fef3f76046efddc810365abcb7d). But it still fails. So I revert it 😅 .

Reply:
Context manager trick works nicely! Thanks for the cool idea @BSchilperoort ! 👍